### ISSUE TITLE
Clarify "sandboxed" in documentation

### ISSUE THREAD

**Issue by koenvervloesem** [AppDaemon/appdaemon#974]
The documentation says that AppDaemon is a "sandboxed python execution environment". But I can't find any explanation about what is sandboxed about AppDaemon. If I search for "sandbox" in the documentation, I only get [two results](https://appdaemon.readthedocs.io/en/latest/search.html?q=sandbox), and if I search for the same term in the GitHub project, I also get only [two results](https://github.com/home-assistant/appdaemon/search?q=sandbox). They all seem to just repeat this sentence.

How are these Python apps isolated from each other exactly? Can this be documented more thoroughly?

**Comment by ReneTode** [comment]
all apps run in their own thread (unless you change default settings)

its hard to tell how apps are isolated, because they are not really.
with AD you create buildingblocks that can work as a single block or can be combined to a bigger whole.

**Comment by koenvervloesem** [comment]
Ok, so does this mean all apps share the same Python libraries? So if:

- app 1 has foobar==1.0.1 in its requirements.txt
- app 2 has foobar==1.0.2 in its requirements.txt

Then this setup fails?

**Comment by ReneTode** [comment]
i think thats in general for any python environment.

an app doesnt have a requirements.txt by the way.
appdaemon has it, but not a single app.

**Comment by koenvervloesem** [comment]
Ok that's partly why I was confused about the "sandboxed" part, because I did see `requirements.txt` files in app directories, but they are all just lumped together if I'm reading [this](https://appdaemon.readthedocs.io/en/latest/DOCKER_TUTORIAL.html?highlight=requirements.txt#adding-dependencies) now.

I think it's good to clarify this somewhere in the documentation.

**Comment by ReneTode** [comment]
i dont know what you mean with lumped together, but if you read the whole docs you will know how appdaemon works. and you can also look at tutorials (mentioned in the docs) or examples (mentioned in the docs)

its hard to explain anything if people assume something based on a single file they find somewhere.

**Comment by koenvervloesem** [comment]
I haven't "assumed something based on a single file", I have read almost the complete documentation and have written a couple of AppDaemon apps, that's why I was asking this question ;-) It just felt weird to see "sandboxed" in the introduction of AppDaemon when I didn't find any proof in the whole documentation I read that there's something what I consider a "sandbox" (with technology such as seccomp, cgroups, ... something container-like) in AppDaemon. That was also why I was asking about the requirements.txt, to see whether I really wasn't missing something.

To sum up: this was just a question to clarify the documentation, I'm just trying to help this awesome project (like in https://github.com/home-assistant/appdaemon/issues/970). If you don't consider it necessary to clarify the documentation, feel free to close this issue.

**Comment by ReneTode** [comment]
im not a dev (anymore) ;) 

i had never given the word sandboxed a second thought. i dont know which definition of sandboxed made Andrew use it. its there already for over 4 years and untill now i have never heard anyone asking about it ;) 

i just did lookup sandboxed python and didnt find anything that i consider related to AD, but i dont think that that 1 word in the small description makes AD different then it is.

i just misinterpreted that you didnt understand AD (or python) at all, because you asked if using 2 versions from 1 lib would fail. which will always fail, because of how python works.
and it doesnt matter how you describe AD, but it is a python program, so it only can do what python can do.

but im sure that Andrew reads this at some point and will consider if he wants to rewrite that introduction, or clarify it.
